docs/fsharp/style-guide/conventions.md

```diff
@@ -437,11 +437,118 @@ Finally, automatic generalization is not always a boon for people who are new to
 
 ## Performance
 
+### Prefer structs for small data types
+
+Using structs (also called Value Types) can often result in higher performance for some code because it typically avoids allocating objects. However, structs are not always a "go faster" button: if the size of the data in a struct exceeds 16 bytes, copying the data can often result in more CPU time spend than using a reference type.
+
+To determine if you should use a struct, consider the following two points:
+
+1. If the size of your data is 16 bytes or smaller
+2. If you are likely to have many of these data types resident in memory in a running program
+
+If the first point applies, you should generally use a struct. If both apply, you should almost always use a struct.
+
```

Comment:
Should you explain why almost always and not always?

Reply:
Yep, did. Gist of it is that if you measure, you could still find no difference or perhaps structs being worse. So it's always important to measure.